### PR TITLE
fix(extension): [lw-10789] fix delegation tx summary

### PR DESCRIPTION
**Pull request by vetalcore**
# Checklist

- [x] JIRA - [LW-10789](https://input-output.atlassian.net/browse/LW-10789)
- [ ] Proper tests implemented
- [ ] Screenshots added.

---

## Proposed solution

Explain how does this PR solves the problem stated in JIRA ticket.
You can also enumerate different alternatives considered while approaching this task.

## Testing

Describe here, how the new implementation can be tested.
Provide link or briefly describe User Acceptance Criteria/Tests that need to be met

## Screenshots

Attach screenshots here if implementation involves some UI changes

[LW-10789]: https://input-output.atlassian.net/browse/LW-10789?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/staking/src/features/overview/helpers/hasPendingDelegationTransaction.ts [hunk 1]
```diff
@@ -1,7 +1,12 @@
...
+import { ActivityStatus, AssetActivityListProps, ConwayEraCertificatesTypes, DelegationActivityType } from '@lace/core';
 import flatMap from 'lodash/flatMap';
 
 export const hasPendingDelegationTransaction = (walletActivities: AssetActivityListProps[]) =>
   flatMap(walletActivities, ({ items }) => items).some(
-    ({ type, status }) => type && type in DelegationActivityType && status === ActivityStatus.PENDING
+    ({ type, status }) =>
+      type &&
+      (type in DelegationActivityType ||
+        type === ConwayEraCertificatesTypes.Registration ||
```
Comment (mchappell) [inline packages/staking/src/features/overview/helpers/hasPendingDelegationTransaction.ts:9]: why not add to the `DelegationActivityType` ?

**Comment by vetalcore** [comment]
i've run into the case when `mapWalletActivities` util returns cached results in case there is new pending tx due to resolve function checking only amount of inFlight transactions.

[`wallet-actiities-slice.ts`](https://github.com/input-output-hk/lace/blob/main/apps/browser-extension-wallet/src/stores/slices/wallet-activities-slice.ts#L366)
```
  (
    { addresses, transactions, assetInfo, delegation: { rewardsHistory } },
    { cardanoFiatPrice, fiatCurrency, assetId },
    { cardanoCoin, assetDetails }
  ) =>
    `${transactions.history.length}_${transactions.outgoing.inFlight.length}_${assetInfo.size}_${
      rewardsHistory.all.length
    }_${cardanoFiatPrice}_${fiatCurrency.code}_${assetId || ''}_${cardanoCoin?.id}_${assetDetails?.id}_${
      addresses[0]?.address
    }`
```
@mkazlauskas i'm wondering if there was a particular performance degradation you were trying to fix by introducing memoization here, or i could safely remoe it for now?
@mchappell @ljagiela fyi.

File: packages/staking/src/features/overview/helpers/hasPendingDelegationTransaction.ts (hunk 1 above)
Comment (vetalcore) [reply packages/staking/src/features/overview/helpers/hasPendingDelegationTransaction.ts:9]: yes an no, these are still conway era certs

**Comment by mkazlauskas** [comment]
> @mkazlauskas i'm wondering if there was a particular performance degradation you were trying to fix by introducing memoization here, or i could safely remove it for now?

Don't remember for sure, but I think yes there was. If I understand the issue correnctly, maybe you could do sth like `transactions.outgoing.inFlight.map(({id}) => id).join()`

**Pull request merged**